modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -423,6 +424,20 @@ public void visit(BLangFunctionTypeNode functionTypeNode) {
         resultType = bInvokableType;
     }
 
+    /**
+     * Lookup all the visible symbols for a given environment scope
+     * @param env Symbol environment
+     * @return all the visible symbols
+     */
+    public Map<Name, ScopeEntry> lookupAllVisibleSymbols(SymbolEnv env) {
```

Comment:
Can you please explain why you need this? Btw, the method name is incorrect. It should be getAllInScopeSymbols or similar

Reply:
This is for Language server symbol resolving. When we identify a certain scope, where the cursor is, and then we need to find all the visible symbols for that scope, including the parent scopes' symbols as well. Will change the method name